lang/values.md

```diff
@@ -181,19 +181,19 @@ Note that types like `&!` have no valid value: when the pointee type is uninhabi
 ```rust
 impl Type {
     fn decode<M: Memory>(Type::Tuple { fields, size }: Self, bytes: List<AbstractByte<M::Provenance>>) -> Option<Value<M>> {
-        if bytes.len() != size { throw!(); }
+        if size.bytes() != bytes.len() { throw!(); }
         Value::Tuple(
             fields.iter().map(|(offset, ty)| {
-                ty.decode::<M>(bytes[offset..][..ty.size::<M>()])
+                ty.decode::<M>(bytes[offset.bytes()..][..ty.size::<M>().bytes()])
             }).collect()?,
         )
     }
     fn encode<M: Memory>(Type::Tuple { fields, size }: Self, val: Value<M>) -> List<AbstractByte<M::Provenance>> {
         let Value::Tuple(values) = val else { panic!() };
         assert_eq!(values.len(), fields.len());
-        let mut bytes = list![AbstractByte::Uninit; size];
+        let mut bytes = list![AbstractByte::Uninit; size.bytes()];
         for ((offset, ty), value) in fields.iter().zip(values) {
-            bytes[offset..][..ty.size::<M>()].copy_from_slice(ty.encode::<M>(value));
+            bytes[offset.bytes()..][..ty.size::<M>().bytes()].copy_from_slice(ty.encode::<M>(value));
```

Comment:
I actually don't like how much `.bytes()` we need to spam sometimes.
But I cannot implement the `List` in a way that understands `Size`,
except if we implement `Deref<Target=BigInt>` or some other trait.
But that would kinda defeat the purpose of the Size newtype.

Reply:
Can `prelude/size.md` implement `Index<RangeFrom<Size>> for List<T>` and `Index<RangeTo<Size>> for List<T>`?

Reply:
Ah but that would assume all lists are byte-indexed, not bit-indexed... :/